.github/workflows/msrv.yml

```diff
@@ -0,0 +1,20 @@
+name: Verify msrv
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  CARGO_INCREMENTAL: 0
+  RUSTFLAGS: "-Dwarnings"
+
+jobs:
+  msrv:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: taiki-e/install-action@cargo-hack
+    - run: cargo hack check --rust-version --workspace --all-targets --ignore-private && diff Cargo.lock Cargo.lock.msrv
```

Comment:
I feel like this is going to fail quickly. If any dependency releases a new crate version that is MSRV-compatible, the `Cargo.lock` created here will differ from the checked-in `Cargo.lock.msrv`. Or am I missing something here?